doc/api/crypto.md

```diff
@@ -1363,7 +1363,7 @@ rapidly.
 
 In line with OpenSSL's recommendation to use PBKDF2 instead of
 [`EVP_BytesToKey`][] it is recommended that developers derive a key and IV on
-their own using [`crypto.pbkdf2()`][] and to use [`crypto.createCipheriv()`][]
```

Comment:
Why are we preferring scrypt over pbkdf2? This is the first I've heard of the recommendation

Reply:
It needs to a simple recommendation - just _do this_, not _do this OR do that_ - because otherwise you need to explain the trade-offs to readers that aren't cryptographers and probably only glance at the documentation anyway.

When it's between scrypt and pbkdf2, I hope we can agree scrypt > pbkdf2.

Reply:
> When it's between scrypt and pbkdf2, I hope we can agree scrypt > pbkdf2.

I think we can agree that scrypt > pbkdf2 for people who don't have highly specific requirements.

Reply:
@benjamingr As scrypt involves a specific (non-customizable) instance of PBKDF2 which is believed to be secure, we can assume that it is at least as computationally secure as PBKDF2 itself, with the "benefit" of preventing any security-violating modifications by the user (e.g. using SHA1 and the like). One "problem" with PBKDF2 is that it can in theory be cracked using enough parallelization (e.g. using GPUs which tend to have many parallel cores), whereas scrypt requires a non-trivial amount of memory to derive a key which heavily limits parallel computation on such devices.